Domain.FSharp/Shipping/Location.fs

```diff
@@ -0,0 +1,24 @@
+namespace FSharp.Domain.Shipping.Location
+
+open System
+open System.Text.RegularExpressions
+
+type UnLocode(value: string) =
+
+    do
+        if isNull value then raise (ArgumentNullException("value"))
+
+    do
+        let pattern = Regex "[a-zA-Z]{2}[a-zA-Z2-9]{3}"
+        let matches = pattern.Match value
+        if not matches.Success then raise (ArgumentException("Provided value is not a valid UnLocode", "value"))
+
+    member val Value = value
+
+type Location(unlocode: UnLocode, name: string) =
+
+    do
+        if String.IsNullOrWhiteSpace(name) then raise (ArgumentException("Provided name is not valid", "name"))
```

Comment:
```suggestion
        if String.IsNullOrWhiteSpace(name) then raise (ArgumentException ("Provided name is not valid", "name"))
```

Style nit: As commented earlier.

FWIW I've only seen this style documented [here](https://blog.ploeh.dk/2018/01/29/argument-list-isomorphisms/#97ea8f87723f4b1bb995b0824b1caba4) (make sure to read the whole section). See also this style in action elsewhere, e.g. https://github.com/hedgehogqa/fsharp-hedgehog/blob/master/src/Hedgehog/Seed.fs